jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -320,8 +320,13 @@ private void joinThreads(long stopByNanos) throws InterruptedException
             long canWait = TimeUnit.NANOSECONDS.toMillis(stopByNanos - System.nanoTime());
             if (LOG.isDebugEnabled())
                 LOG.debug("Waiting for {} for {}", thread, canWait);
-            if (canWait > 0)
-                thread.join(canWait);
+            if (canWait > 0) {
+                try {
+                    thread.join(canWait);
+                } catch (InterruptedException e) {
+                    // ignore
```

Comment:
It was interrupted, which means the thread still exists, and has the interrupted flag set.
This seems to bypass the `interrupted()` cleanup in the QueuedThreadPool.
Not 100% sold on this.
I appreciate that you have protection so that 1 bad `Thread.join()` doesn't prevent other threads from completing, but there's something off about this approach. (Can't put my finger on it)

I think, historically speaking, this is intended to be a speculative join, where it's allowed to fail.  But I could be misremembering things.

Reply:
I'm onboard with handling interruptedException here, but I agree with @joakime that ignoring it does not look correct.
Perhaps we should just join on the thread again?   We could do this in a loop or once only?  

Eitherway, we'd need a unit test in this PR.  @kohlschuetter do you wish to keep working on this or do  you want us to take it over?

Reply:
@gregw Feel free to take over from here, thanks!